src/feasibility-form-nls.jl

```diff
@@ -39,38 +39,41 @@ end
 Converts a nonlinear least-squares problem with residual `F(x)` to a nonlinear
 optimization problem with constraints `F(x) = r` and objective `¹/₂‖r‖²`.
 """
-function FeasibilityFormNLS(nls::AbstractNLSModel; name = "$(nls.meta.name)-ffnls")
+function FeasibilityFormNLS(
+  nls::AbstractNLSModel{T, S};
+  name = "$(nls.meta.name)-ffnls",
+) where {T,S}
   nequ = nls.nls_meta.nequ
   meta = nls.meta
   nvar = meta.nvar + nequ
   ncon = meta.ncon + nequ
   nnzh = nls.nls_meta.nnzh + nequ + (meta.ncon == 0 ? 0 : meta.nnzh) # Some indexes can be repeated
-  meta = NLPModelMeta(
+  meta = NLPModelMeta{T, S}(
     nvar,
-    x0 = [meta.x0; zeros(nequ)],
-    lvar = [meta.lvar; fill(-Inf, nequ)],
-    uvar = [meta.uvar; fill(Inf, nequ)],
+    x0 = [meta.x0; zeros(T, nequ)],
```

Comment:
I think this fails for GPU but I don't see a viable alternative, do you?

Reply:
Would this work?
```julia
x0= zero(meta.x0, meta.nvar + nequ)
x0[1 : meta.nvar] .= meta.x0
```

Reply:
I didn't test it, but this [tutorial](https://nextjournal.com/sdanisch/julia-gpu-programming) suggests that `cat` and `fill` are available methods for GPUArrays.